src/core/elements/getElements.js

```diff
@@ -31,17 +31,29 @@ const downloadElements = async (elements, query, jobId, processId, isPrivate, jo
         return null;
       }
 
-      logDebug(`Downloading element for element key - ${element.key}`, jobId);
-      const exportedElement = await http.get(makePath(element), query, account);
-      logDebug(`Downloaded element for element key - ${element.key}`, jobId);
-
       emitter.emit(EventTopic.ASSET_STATUS, {
         processId,
         assetType: Assets.ELEMENTS,
         assetName: element.key,
-        assetStatus: equals(jobType, JobType.PROMOTE_EXPORT) ? ArtifactStatus.INPROGRESS : ArtifactStatus.COMPLETED,
+        assetStatus: ArtifactStatus.INPROGRESS,
         metadata: elementMetadata,
       });
+
+      logDebug(`Downloading element for element key - ${element.key}`, jobId);
+      const exportedElement = await http.get(makePath(element), query, account);
+      logDebug(`Downloaded element for element key - ${element.key}`, jobId);
+
+      // If 'promote_export' job, the final artifact
+      // status update will happen in doctor-service
+      if (!equals(jobType, JobType.PROMOTE_EXPORT)) {
+        emitter.emit(EventTopic.ASSET_STATUS, {
+          processId,
+          assetType: Assets.ELEMENTS,
+          assetName: element.key,
+          assetStatus: ArtifactStatus.COMPLETED,
```

Comment:
we need the promote condition here

Reply:
The promote condition is there in line 48.
If it's a promote job, we simply don't update the status to `completed`